lib/Conversion/ArithExtToArith/ArithExtToArith.td

```diff
@@ -0,0 +1,36 @@
+#ifndef LIB_CONVERSION_ARITHEXTTOARITH_ARITHEXTTOARITH_TD_
+#define LIB_CONVERSION_ARITHEXTTOARITH_ARITHEXTTOARITH_TD_
+
+include "lib/DRR/Utils.td"
+include "lib/Dialect/ArithExt/IR/ArithExtOps.td"
+include "mlir/Dialect/Arith/IR/ArithOps.td"
+include "mlir/IR/PatternBase.td"
+include "mlir/Pass/PassBase.td"
+
+def ArithExtToArith : Pass<"arith-ext-to-arith", "ModuleOp"> {
+  let summary = "Lower `arith_ext` to standard `arith`.";
+
+  let description = [{
+    This pass lowers the `arith_ext` dialect to their `arith` equivalents.
+  }];
+
+  let dependentDialects = [
+    "mlir::arith::ArithDialect",
+    "mlir::heir::arith_ext::ArithExtDialect",
+  ];
+}
+
+// Using DRR to generate the lowering patterns for specific operations
+
+defvar DefGE = ConstantEnumCase<Arith_CmpIPredicateAttr, "uge">;
```

Comment:
I wonder if we will always want this to be unsigned (uge = unsigned greater equal) vs signed. I feel that perhaps we should all agree on the semantics of arith_ext ops on this front, i.e., that we always take the representative in [0, modulus) for modular operations.

Still happy to submit and revisit this later.

Reply:
I agree we should discuss this further.

I think if we let it be signed then we will have similar confusion/issue with using RemSI where it treats both operands as signed and we generally don't want the second operand to be signed.

FWIW, in the context of HEaan, they do make the assumption that it is an unsigned input.

Reply:
I was planning to spend some time in the next meeting on discussing how we model modular arithmetic, and data-type semantics in general, and I think the signedness question would be a good addition to the list :)